text/0000-cargo-miri-integration.md

```diff
@@ -0,0 +1,146 @@
+
+Commands are translated as follows:
+
+```sh
+cargo foo
+```
+
+becomes
+
+```sh
+corgi $PATH_TO_FOO -- /path/to/your/crate/main.rs
+```
+
+where `$PATH_TO_FOO` is either the path specified in `Cargo.toml` or `<sysroot>/cargo/bin/foo/main.rs` if unspecified.
+
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+The slowness of `miri` may increase build times mildly. However, this expense is deemed small in comparison to the vast might of safe-build-system-yness that will be unleashed on a UB-impoverished world.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+> - Why is this design the best in the space of possible designs?
+
+Seems obvious, right? `miri`.
+
+> - What other designs have been considered and what is the rationale for not choosing them?
+
+- Other virtualization techniques could be used, including virtual machines or containers. Such a design would increase sandboxing of build scripts, but it doesn't check for UB or increase the customizability of builds.
+- Custom wrappers around `cargo` can customize the build process. `cargo` "pluggins" are one means of doing this (e.g. `rust-analyzer` uses the `xtask` pluggin system). Other projects opt to write their own build systems (e.g. `rustc` itself). Still others, just wrap `cargo` in `Makefile`s, `bazel`, or other more flexible build systems. These options have many disadvatages:
+    - They are clunky and non-standard. They require the user to ignore muscle memory and do something custom.
+    - They can execute even more code without a sandbox.
+    - They often require reimplementing features of `cargo` outside of `cargo`.
+
+> - What is the impact of not doing this?
+
+It is highly likely that the world will end due to UB. Or at best my laundry might get eaten.
+
+# Prior art
+[prior-art]: #prior-art
+
+There is so much prior work, I wouldn't know where to start. A quick search for "virtualization"  on ACM Digital Library returns 126753 results. A search for "interpretter" returns 158217 results. I haven't read all of them, but my guess is that the ones I haven't read all support the ideas of this RFC.
+
+There have been many [discussions on IRLO about sandboxing code](https://internals.rust-lang.org/search?q=sandbox).
+
+Ken Thompson's excellent [Reflections on Trusting Trust](https://dl.acm.org/doi/10.1145/358198.358210) underscores the importance of trusting one's infrastructure. As we all know, if a program is UB-free, it is entirely trustworthy.
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- Are corgis really the best dog to name this tool after? The name at least sounds like `cargo` a bit...
+- What about all the cat-lovers in the rust community?
```

Comment:
Who needs 'em? This is a non-issue, please remove it from the unresolved questions as it's quite self-evident

Reply:
:cat2:

Reply:
>What about all the cat-lovers in the rust community?

[Miri is a cat](https://www.youtube.com/watch?v=d8zXlwEijN4) in the first place, so their needs are already covered.